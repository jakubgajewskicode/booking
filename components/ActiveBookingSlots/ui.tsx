import styled from 'styled-components';

import { device } from '../../utils/media';

const ItemContainer = styled.div`
  grid-column: span 8;
  text-align: center;
  cursor: pointer;
  transition: opacity 0.7s ease;

  @media ${device.mobileL} {
    grid-column: span 5;
  }

  @media ${device.tablet} {
    grid-column: span 4;
  }

  :hover {
    opacity: 0.5;
  }
`;

export { ItemContainer };
