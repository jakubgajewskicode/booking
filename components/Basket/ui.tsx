import styled from 'styled-components';

const StyledSection = styled.section`
  display: flex;
  padding: 20px;
  padding: 20px 20px;
  justify-content: space-around;

  grid-row-start: 1;
`;

const ButtonWrapper = styled.div`
  width: 100%;
  text-align: center;
`;

const Div = styled.div`
  text-align: center;
`;

const Title = styled.div`
  text-align: center;
`;

const Flex = styled.div`
  display: flex;
`;

const SlotsWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  width: 20%;
`;

const WorkerDetails = styled.div`
  width: 80%;
`;

export {
  StyledSection,
  ButtonWrapper,
  Div,
  Title,
  Flex,
  SlotsWrapper,
  WorkerDetails,
};
