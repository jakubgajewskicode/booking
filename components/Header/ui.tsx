import styled from 'styled-components';

import { device } from '../../utils/media';

export const StyledHeader = styled.header`
  background-color: #ebebeb;
  height: 15vh;
  position: relative;
`;

export const H1 = styled.h1`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  margin: 0;
  font-size: 20px;

  @media ${device.tablet} {
    font-size: 35px;
  }
`;

export const LogoWrapper = styled.div`
  width: 10%;
  height: 100%;
  position: relative;

  svg {
    position: absolute;
    top: -10%;
    left: 10%;
    width: 70px;
    height: 100px;
  }

  @media ${device.tablet} {
    width: 5%;
    svg {
      width: 120px;
      top: 40%;
    }
  }
`;
