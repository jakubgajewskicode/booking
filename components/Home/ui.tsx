import styled from 'styled-components';

import { device } from '../../utils/media';

export const Form = styled.form`
  text-align: center;
  padding-bottom: 15px;
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin-top: 25px;
`;

export const Input = styled.input`
  width: 100%;
  height: 45px;
  border: 1px solid #ebebeb;
  box-shadow: 0 3px 10px rgb(0 0 0 / 0.2);
`;

export const Grid = styled.section`
  @media ${device.tablet} {
    display: grid;
    grid-template-columns: repeat(12, 1fr);
    grid-template-rows: 35% 45% 100%;
  }
`;

export const PropertyTypes = styled.aside`
  grid-column: span 2;
  grid-row-start: 3;
  margin: 10px 10px 10px 20px;

  @media ${device.tablet} {
  }
`;

export const SearchSection = styled.section`
  grid-column: 4 / span 10;
  margin-top: 55px;
`;

export const StyledUl = styled.ul`
  list-style: none;
  padding: 0;
  background-color: #ebebeb;
  padding: 10px;
`;

export const StyledLi = styled.li`
  cursor: pointer;
  margin: 10px 10px 10px 0;

  :hover {
    font-weight: bold;
  }
`;

export const Label = styled.label`
  margin: 0;
`;

export const StyledLabel = styled.label`
  margin: 0;
  font-weight: 600;
`;

export const SearchResults = styled.section`
  grid-column: 4 / span 10;
`;

export const Wrapper = styled.div``;

export const SubGrid = styled.div`
  display: flex;

  div:first-child {
    width: 30%;
  }
`;

export const StyledResultLabel = styled.div`
  width: 100%;
  text-align: center;
`;

export const Table = styled.table`
  border-collapse: collapse;
  width: 100%;
  margin-top: 25px;
`;

export const TH = styled.th`
  padding: 10px;
  text-align: left;
  border: 1px solid #ebebeb;
  text-align: center;
  background-color: #ebebeb;
  border-right: 3px solid #ffffff;
`;

export const TD = styled.td`
  padding: 0.25rem;
  text-align: center;
  border-top: 1px solid #ebebeb;
  position: relative;
  padding: 10px;
`;

export const TickWrapper = styled.div`
  width: 15px;
  margin: auto;
`;

export const SelectedProperties = styled.section`
  grid-column: 4 / span 10;
  display: flex;
  flex-direction: column;
`;

export const MainSection = styled.section`
  grid-column: 4 / span 10;
`;

export const ButtonContainer = styled.div`
  width: 50%;
`;

export const StyledTBody = styled.tbody`
  background-color: #f4faff;
`;

export const StyledCheckbox = styled.input`
  cursor: pointer;
`;
