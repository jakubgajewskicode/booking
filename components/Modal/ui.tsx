import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`from { opacity: 0; }`;

export const Container = styled.div`
  position: relative;
  z-index: 0;
`;
export const Overlay = styled.div`
  animation: ${fadeIn} 200ms ease-out;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90vh;
  height: 50vh;
  outline: 0;
  z-index: 1000;
  box-shadow: 6.4px 12.8px 12.8px hsl(0deg 0% 0% / 0.3);
`;
export const Dialog = styled.div`
  background: #dbdbdb;
  margin: 0;
  padding-bottom: 30px;
  position: relative;
  overflow: auto;
  height: 50vh;
`;
export const Close = styled.a`
  cursor: pointer;
  z-index: 2;
  position: absolute;
  top: 32px;
  right: 32px;
`;
