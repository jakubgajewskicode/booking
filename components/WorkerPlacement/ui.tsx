import styled from 'styled-components';

const List = styled.ul`
  grid-column: span 4;
  text-align: center;
  list-style: none;
`;

const SectionSubHeader = styled.div`
  text-align: center;
`;

const ThanksMsg = styled.div`
  height: 50vh;
  position: relative;
`;

const StyledP = styled.p`
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  position: absolute;
`;

export { List, SectionSubHeader, ThanksMsg, StyledP };
