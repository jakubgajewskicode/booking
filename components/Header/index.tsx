import { H1, LogoWrapper, StyledHeader } from './ui';

import Logo from './logo';

const Header = () => (
  <StyledHeader>
    <LogoWrapper>
      <Logo />
    </LogoWrapper>
    <H1>Property search tool</H1>
  </StyledHeader>
);

export default Header;
