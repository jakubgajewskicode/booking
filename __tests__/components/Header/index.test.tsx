import { shallow } from 'enzyme';

import Header from '../../../components/Header';
import { H1 } from '../../../components/Header/ui';

describe('Header renders', () => {
  test('renders', () => {
    const header = shallow(<Header />);
    const H1Wrapper = header.find(H1);

    expect(header).toMatchSnapshot();
    expect(H1Wrapper).toHaveLength(1);
    expect(H1Wrapper.text()).toEqual('Property search tool');
  });
});
