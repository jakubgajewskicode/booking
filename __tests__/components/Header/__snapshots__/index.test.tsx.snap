// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders renders 1`] = `
<ui__StyledHeader>
  <ui__LogoWrapper>
    <Logo />
  </ui__LogoWrapper>
  <ui__H1>
    Property search tool
  </ui__H1>
</ui__StyledHeader>
`;
